.github/workflows/tests.yaml

```
@@ -0,0 +1,23 @@
+name: Tests
+
+on:
+  push:
+  pull_request:
+
+permissions:
+  contents: read
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: Install Go
+      uses: actions/setup-go@v5
+      with:
+        go-version-file: 'go.mod'
+
+    - run: make test
```

Maybe add validate and build steps right away?

```suggestion
    - name: Validate
      uses: golangci/golangci-lint-action@v4

    - name: Test
      run: make test

    - name: Build
      run: make build
```

---

We'd need to fix existing linting issues (which is a good thing) though. It can be done in a separate PR that should be merged ahead of this one.

---

I was thinking of doing the reverse, adding test for CI first since it's easy to add and provides value right away, and later add validation (eg: linting) to CI with their respective fix.

For example, a lot of lint issues are `Error return value of X is not checked`. I could either do the error check, which imo is a good thing, but potentially introduce breaking change OR "fix" those lint issues by explicitely ignoring the error (not in favor).

WDYT?

CC: @maxsokolovsky @JonCrowther

---

> I was thinking of doing the reverse
Either way works IMHO 👍